### ISSUE TITLE
Problems with ITV Cache disable GTA

### ISSUE THREAD

**Issue by Themanwithoutaplan** [Ascoware/get-iplayer-automator#210]
If there is, for any reason, a problem updating the ITV cache then GTA essentially locks up because the search can't be used. Would it be possible to add a timeout for cache updates?

**Comment by skovatch** [comment]
Yes, I have noticed this, and yes, it could be better. The web site was out of service over the weekend and I noticed it then.

**Comment by BruceGoatly** [comment]
This seems to be an issue again this morning. (The obvious workaround is to disable caching ITV programmes, but it's a bit disconcerting when GiA locks up...)

**Comment by skovatch** [comment]
I think I can cut it back to a 10 second timeout from the default of 7 days(!) for resource requests. The right thing to do, according to Apple, is to add a Cancel button so you can interrupt the whole process if it's taking too long, but it seems like 10 seconds will do for now until I can find a good way to put a cancel button into the UI.

**Comment by skovatch** [comment]
I'm going to see if I can work on one or two other things while I'm on break before pushing out a release.

**Comment by hugofalkman** [comment]
@skovatch 10 seconds is too short, we are not all on hi speed lines. Suggest 30 seconds unless somebody else feels an even longer timeout is necessary.
